tlsproxy.go

```diff
@@ -158,3 +158,55 @@ func doRun(idleTimeout time.Duration, l net.Listener, dial func() (net.Conn, err
 		}()
 	}
 }
+
+func dial(keepAlivePeriod time.Duration, forwardAddr string) (net.Conn, error) {
+	conn, err := net.DialTimeout("tcp", forwardAddr, 30*time.Second)
+	if err == nil {
+		addKeepalive(keepAlivePeriod, conn)
+	}
+	return conn, err
+}
+
+func addKeepalive(keepAlivePeriod time.Duration, conn net.Conn) net.Conn {
+	if keepAlivePeriod <= 0 {
```

Comment:
`wrapKeepAliveListener` already handles the case when `keepAlivePeriod` <= 0. This check can be moved up to `dial`.